kayenta-stackdriver/src/test/java/com/netflix/kayenta/stackdriver/metrics/StackdriverMetricsServiceTest.java

```diff
@@ -0,0 +1,153 @@
+
+  @Mock GoogleNamedAccountCredentials stackdriverCredentialsMock;
+  @Mock ListMetricDescriptorsResponse listMetricDescriptorsResponseMock;
+  @Mock StackdriverCanaryScope canaryScopeMock;
+
+  @Mock(answer = RETURNS_DEEP_STUBS)
+  Monitoring monitoringMock;
+
+  @Mock Monitoring.Projects.TimeSeries.List timeSeriesListMock;
+  @Mock ListTimeSeriesResponse responseMock;
+
+  @Mock(answer = RETURNS_DEEP_STUBS)
+  Registry registry;
+
+  @InjectMocks StackdriverMetricsService stackdriverMetricsService;
+
+  @Test
+  public void readsInt64Metrics() throws IOException {
+    when(accountCredentialsRepoMock.getRequiredOne(ACCOUNT)).thenReturn(stackdriverCredentialsMock);
+
+    when(stackdriverCredentialsMock.getMonitoring()).thenReturn(monitoringMock);
+
+    when(monitoringMock
+            .projects()
+            .timeSeries()
+            .list(anyString())
+            .setAggregationAlignmentPeriod(anyString())
+            .setAggregationCrossSeriesReducer(anyString())
+            .setAggregationPerSeriesAligner(anyString())
+            .setFilter(anyString())
+            .setIntervalStartTime(anyString())
+            .setIntervalEndTime(anyString()))
+        .thenReturn(timeSeriesListMock);
+
+    when(timeSeriesListMock.execute()).thenReturn(responseMock);
+
+    List<TimeSeries> timeSeriesListWithInt64Points = new ArrayList<TimeSeries>();
+
+    // Create a time series with INT64 points
+    List<Point> int64Points = new ArrayList<Point>();
+    int64Points.add(
+        new Point()
+            .setValue(new TypedValue().setInt64Value((Long) 64l))
+            .setInterval(
+                new TimeInterval()
+                    .setStartTime("1970-01-01T00:00:00.00Z")
+                    .setEndTime("1970-01-01T00:00:01.00Z")));
+    TimeSeries timeSeriesWithInt64Points =
+        new TimeSeries().setValueType("INT64").setPoints(int64Points);
+    timeSeriesListWithInt64Points.add(timeSeriesWithInt64Points);
+
+    when(responseMock.getTimeSeries()).thenReturn(timeSeriesListWithInt64Points);
+
+    CanaryConfig canaryConfig = new CanaryConfig();
+    CanaryMetricConfig canaryMetricConfig =
+        CanaryMetricConfig.builder()
+            .name("metricConfig")
+            .query(
+                StackdriverCanaryMetricSetQueryConfig.builder()
+                    .resourceType("global")
+                    .metricType("instance")
+                    .build())
+            .build();
+
+    CanaryScope canaryScope =
+        new StackdriverCanaryScope()
+            .setStart(Instant.EPOCH)
+            .setEnd(Instant.EPOCH.plusSeconds(1))
+            .setStep(1l);
+    List<MetricSet> queriedMetrics =
+        stackdriverMetricsService.queryMetrics(
+            ACCOUNT, canaryConfig, canaryMetricConfig, canaryScope);
+
+    assertThat(queriedMetrics.get(0).getValues()).contains(64d);
+  }
+
+  @Test
+  public void returnsSingleMetricDescriptorInCache() throws IOException {
```

Comment:
This test passes without the code change above.  Can you help me understand what it's for?

Reply:
It was mainly for me working my way up to writing the actual test for my change. It was a simpler test to write so I started with it to make sure I understood correctly how Mockito works.

In short, it's not related to my change but I figured since I already written it I might as well add it. It can be removed or included in a different PR to keep things relevant. Let me know, I won't take it personally :)

Reply:
Let's keep it!